metricbeat/module/system/raid/raid.go

```diff
@@ -67,33 +69,62 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 		return nil, err
 	}
 
+	sysMountPoint := filepath.Join(config.MountPoint, "/sys/block")
+
 	return &MetricSet{
 		BaseMetricSet: base,
 		fs:            fs,
+		sysblock:      sysMountPoint,
 	}, nil
 }
 
+func blockto1024(b int64) int64 {
+	//Annoyingly, different linux subsystems report size size using different blocks
+	// /proc/mdstat /proc/partitions and mdadm (Via the BLKGETSIZE64 ioctl call) count "size" as the number of 1024-byte blocks.
+	// /sys/block/md*/size uses 512-byte blocks. As does /sys/block/md*/md/sync_completed
+	//convert the 512-byte blocks to 1024 byte blocks to maintain how this metricset "used to" report size
+	return b / 2
+}
+
 // Fetch fetches one event for each device
 func (m *MetricSet) Fetch(r mb.ReporterV2) {
-	stats, err := m.fs.ParseMDStat()
+	devices, err := blockinfo.ListAll(m.sysblock)
 	if err != nil {
-		r.Error(errors.Wrap(err, "failed to parse mdstat"))
+		r.Error(errors.Wrap(err, "failed to parse sysfs"))
 		return
 	}
 
-	for _, stat := range stats {
+	for _, blockDev := range devices {
+
 		event := common.MapStr{
-			"name":           stat.Name,
-			"activity_state": stat.ActivityState,
+			"name":   blockDev.Name,
+			"status": blockDev.ArrayState,
+			"level":  blockDev.Level,
 			"disks": common.MapStr{
-				"active": stat.DisksActive,
-				"total":  stat.DisksTotal,
-			},
-			"blocks": common.MapStr{
-				"synced": stat.BlocksSynced,
-				"total":  stat.BlocksTotal,
+				"active":  blockDev.DiskStates.Active,
+				"total":   blockDev.DiskStates.Total,
+				"spare":   blockDev.DiskStates.Spare,
+				"failed":  blockDev.DiskStates.Failed,
+				"unknown": blockDev.DiskStates.Unknown,
```

Comment:
Oh sorry, I missed this one on the latest reviews. Please don't send the count of unknowns.

Reply:
And remove it from the diskstates struct as is not used anymore.

Reply:
Bah, I totally thought I removed that, my bad!